encoding/descaling.md

```diff
@@ -1,477 +1,332 @@
 
 [getnative]: https://github.com/Infiziert90/getnative
-[descale]: https://github.com/BluBb-mADe/vapoursynth-descale
 
 
-## Finding out the native resolution
+# Filterchains
 
-To figure out what
-the native resolution of an anime is,
-first you need a good frame to test.
-Ideally,
-you'll want a bright frame
-with as little blur as possible of high quality
-(Blu-ray or very good webstreams).
-It also helps to not have
-too many post-processed elements in the picture.
-Whilst it is most definitely possible
-to get pretty good results with "bad" frames,
-it's generally better to use
-good frames whenever possible.
+Once you've figured out the native resolution,
+you can move on to the actual descaling.
+There's two common ways to do this.
+The first one is by using [descale][].
+The other is by using `invks` from [fmtc][].
+For convenience's sake,
+we'll be keeping it to just `descale` for this guide.
 
-Here are some examples of "bad" frames.
+There are a couple very common kernels
+used in anime production:
+* Bilinear
+* Bicubic b=1/3, c=1/3 (Mitchell-Netravali)
+* Bicubic b=0, c=0.5 (Catmull-Rom)
 
-![Manaria Friends — 01 (frame 1)](images/descale_manaria01.png)
+Occasionally, although rare,
+you may also run into the following kernels:
+* Lanczos taps=3
+* Lanczos taps=4
+* Spline16
+* Spline36
+* Bicubic b=1, c=0 (B-Spline)
+* Bicubic b=0, c=0 (Hermite)
+* Bicubic b=0, c=0 (Hermite)
+* Bicubic b=0.3782, c=0.3109 (Robidoux)
+* Bicubic b=0.2620, c=0.3690 (Robidoux Sharp)
+* Bicubic b=0.6796, c=0.1602 (Robidoux Soft)
 
-This picture is dark.
-It also has some effects over it.
+The most reliable way to discern
+which kernel was used is by descaling the clip,
+upscaling it again with the same kernel,
+and then comparing it with the original frame.
+Here is an example script using `compare` from `lvsfunc`
+to easily compare frames from two clips:
 
-![Manaria Friends — 01 (frame 2)](images/descale_manaria02.png)
+```py
+import lvsfunc as lvf
 
-This picture is also very dark
-and has even more effects over it.
+clip = core.lsmas.LWLibavSource(r"PATH/TO/VIDEO.m2ts")
+clip = fvf.Depth(clip, 32) # Important to note: descale requires a GRAYS clip (or 444, but don't do that).
 
-![Manaria Friends — 01 (frame 3)](images/descale_manaria03.png)
+clip_y = core.std.ShufflePlanes(clip, 0, vs.GRAY)
+descaled = core.descale.Debicubic(clip_y, width=1280, height=720, b=1/3, c=1/3)
+reupscaled = core.resize.Bicubic(descaled, width=1920, height=1080, filter_param_a=1/3, filter_param_b=1/3)
+merged = core.std.ShufflePlanes([reupscaled, clip], [0, 1, 2], vs.YUV)
 
-Heavy dynamic grain will almost always give bad results.
+comp = lvf.compare(clip, merged)
+comp.set_output()
+```
 
-![Manaria Friends — 01 (frame 4)](images/descale_manaria04.png)
+The most reliable way to check for differences is
+by zooming in 4x with Nearest Neighbor
+and switching between the frames
+with the left and right arrow buttons.
```

Comment:
Is this using vsedit? Should probably mention that, since yuuno has different means of comparing clips.